x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -87,6 +87,12 @@ export const TakeActionDropdown = React.memo(
     );
     const isEvent = actionsData.eventKind === 'event';
 
+    const isAgentEndpoint = useMemo(() => ecsData?.agent?.type?.includes('endpoint'), [ecsData]);
+
+    const disableEventFilterAction = useMemo(() => {
+      return !isEvent && !isAgentEndpoint;
```

Comment:
I guess this needs to be an OR condition. So this item is disabled if not is event OR not is endpoint. What it means, it has to be enabled if is an event and is endpoint:

- not (isEvent && isEndpoint) => not isEvent || not isEndpoint

Reply:
maybe it would be easier if we do `const isEndpointEvent = isEvent && isAgentEndpoint;` and then set the `disabled={!isEndpointEvent}`, too much double negations are a bit confusing.